ERR_MODULE_NOT_FOUND when running node --loader ts-node/esm src/index.ts

index.ts

```
import { AppDataSource } from "./data-source"
import { User } from "./entity/User"

AppDataSource.initialize().then(async () => {

    console.log("Inserting a new user into the database...")
    const user = new User()
    user.firstName = "Timber"
    user.lastName = "Saw"
    user.age = 25
    await AppDataSource.manager.save(user)
    console.log("Saved a new user with id: " + user.id)

    console.log("Loading users from the database...")
    const users = await AppDataSource.manager.find(User)
    console.log("Loaded users: ", users)

    console.log("Here you can setup and run express / fastify / any other framework.")

}).catch(error => console.log(error))
```

User.ts

```
import { Entity, PrimaryGeneratedColumn, Column, OneToMany } from "typeorm";
import UploadInfo from "./UploadInfo";

@Entity()
export class User {
  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  firstName: string;

  @Column()
  lastName: string;

  @Column()
  age: number;

  @OneToMany((type) => UploadInfo, (uploadInfo) => uploadInfo)
  uploads: UploadInfo[];
}
```

UploadInfo.ts

```
import { Column, Entity, ManyToOne, PrimaryGeneratedColumn } from "typeorm";
import { User } from "./User";

@Entity()
export default class UploadInfo {
  @PrimaryGeneratedColumn()
  id: number;

  @ManyToOne((type) => User, (user) => user.uploads)
  user: User;
}
```

data-source.ts

```
import "reflect-metadata";
import { DataSource } from "typeorm";
import UploadInfo from "./entity/UploadInfo";
import { User } from "./entity/User";

export const AppDataSource = new DataSource({
  type: "sqlite",
  database: "database.sqlite",
  synchronize: true,
  logging: true,
  entities: [User, UploadInfo],
  migrations: [],
  subscribers: [],
});

```

tsconfig.json

```
{
  "compilerOptions": {
    "lib": ["es5", "es6"],
    "target": "esnext",
    "module": "esnext",
    "moduleResolution": "node",
    "outDir": "./build",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "esModuleInterop": true,
    "sourceMap": true
  }
}
```

package.json

```
{
  "type": "module",
  "devDependencies": {
    "@types/node": "^16.11.10",
    "reflect-metadata": "^0.1.13",
    "ts-node": "^10.7.0",
    "typeorm": "^0.3.6",
    "typescript": "4.5.2"
  },
  "dependencies": {
    "pg": "^8.4.0",
    "reflect-metadata": "^0.1.13",
    "sqlite3": "^4.0.3",
    "typeorm": "0.3.6"
  },
  "scripts": {
    "start": "ts-node src/index.ts",
    "typeorm": "typeorm-ts-node-commonjs"
  }
}
```

When running node --loader ts-node/esm src/index.ts, the following error is thrown:

```
(node:27444) ExperimentalWarning: --experimental-loader is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
C:\Users\david\source\testing\typeorm\node_modules\ts-node\dist-raw\node-esm-resolve-implementation.js:383
    throw new ERR_MODULE_NOT_FOUND(
          ^
CustomError: Cannot find module 'C:\Users\david\source\testing\typeorm\src\data-source' imported from C:\Users\david\source\testing\typeorm\src\index.ts
    at finalizeResolution (C:\Users\david\source\testing\typeorm\node_modules\ts-node\dist-raw\node-esm-resolve-implementation.js:383:11)   
    at moduleResolve (C:\Users\david\source\testing\typeorm\node_modules\ts-node\dist-raw\node-esm-resolve-implementation.js:818:10)        
    at Object.defaultResolve (C:\Users\david\source\testing\typeorm\node_modules\ts-node\dist-raw\node-esm-resolve-implementation.js:929:11)    at C:\Users\david\source\testing\typeorm\node_modules\ts-node\src\esm.ts:228:33
    at entrypointFallback (C:\Users\david\source\testing\typeorm\node_modules\ts-node\src\esm.ts:179:34)
    at resolve (C:\Users\david\source\testing\typeorm\node_modules\ts-node\src\esm.ts:227:12)
    at ESMLoader.resolve (node:internal/modules/esm/loader:530:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:251:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:79:40)
    at link (node:internal/modules/esm/module_job:78:36)
```

--- a/src/commands/InitCommand.ts
+++ b/src/commands/InitCommand.ts
@@ -377,7 +377,9 @@ export const Routes = [{
      * Gets contents of the user controller file (used when express is enabled).
      */
     protected static getControllerTemplate(isEsm: boolean): string {
-        return `import { AppDataSource } from "../data-source${isEsm ? ".js" : ""}"
+        return `import { AppDataSource } from "../data-source${
+            isEsm ? ".js" : ""
+        }"
 import { NextFunction, Request, Response } from "express"
 import { User } from "../entity/User${isEsm ? ".js" : ""}"
 

--- a/src/commands/InitCommand.ts
+++ b/src/commands/InitCommand.ts
@@ -377,7 +377,7 @@ export const Routes = [{
      * Gets contents of the user controller file (used when express is enabled).
      */
     protected static getControllerTemplate(isEsm: boolean): string {
-        return `import { AppDataSource } from "../data-source.js"
+        return `import { AppDataSource } from "../data-source${isEsm ? ".js" : ""}"
 import { NextFunction, Request, Response } from "express"
 import { User } from "../entity/User${isEsm ? ".js" : ""}"
 

--- a/src/commands/InitCommand.ts
+++ b/src/commands/InitCommand.ts
@@ -377,7 +377,7 @@ export const Routes = [{
      * Gets contents of the user controller file (used when express is enabled).
      */
     protected static getControllerTemplate(isEsm: boolean): string {
-        return `import { AppDataSource } from '../data-source'
+        return `import { AppDataSource } from "../data-source.js"
 import { NextFunction, Request, Response } from "express"
 import { User } from "../entity/User${isEsm ? ".js" : ""}"
 

